### ISSUE TITLE
Invoke Biometric prompt after a timeout

### ISSUE THREAD

**Issue by AnuragTiwari1** [forcedotcom/SalesforceMobileSDK-ReactNative#272]
Hi, 

I had a requirement where I have to prompt user for pin/biometric auth after inactivity for some time. right now I get the prompt whenever my app goes foreground from background but I need to be able to call the prompt programatically or via adding some setting in connected app.

**Comment by gkotula** [comment]
Hi AnuragTiwari1,

Unfortunately the MobileSDK does not provide APIs for programmatically showing the PIN/biometric lock.  This feature may be considered in the future, but it is not on our roadmap at this time.

**Comment by AnuragTiwari1** [comment]
ok, any suggestion on how should I approach this if I want to create a NativeModule to support this. I think biometric tries to invoke the ScreenLockActivity, I am thinking to invoke this by writing a NativeModule preferably outside the mobile_sdk folder

**Comment by gkotula** [comment]
**TL;DR** The best recommendation I have at this time is to create a workaround solution (such as requiring that devices have a short idle-screen timeout) since the MobileSDK does not officially support manual PIN locking at this time.

---

A couple of things:

Firstly, using internal SDK implementation details to create your own module is not recommended, even though the implementation is open-source and freely browsable.  There is always a risk with relying on internal SDK details to make your own modules since implementation details can change at any time.

By electing to manually control the `ScreenLockActivity` you take on a much higher burden of maintenance to ensure your module stays up-to-date with the internal changes of the SDK.  It is certainly doable _in principle_, but Salesforce may not be able to offer solutions if things break.

The best I can offer at this time is:  If this is critical to your application, I would try thinking of different mechanisms for achieving what you want.  One idea that comes to mind is making your devices have short screen-idle timeouts and just let the SDK naturally lock with the sleep/wake behavior, but likely there are other workarounds as well.

If you really need or want this API publicly-exposed, you can open a pull-request against the native MSDK to add this enhancement.  You would get genuine, direct feedback about the implications of accepting this change to the API, and if accepted you would get exactly what you were looking for.  Of course, we don't do ad-hoc releases for something like this, so even after getting accepted it would be until the next major release to see the changes.
